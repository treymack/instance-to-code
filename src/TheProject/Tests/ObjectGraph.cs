using System;
using System.Linq;
using ApprovalTests;
using NUnit.Framework;

namespace TheProject.Tests
{
    [TestFixture]
    public class ObjectGraph
    {
        [Test]
        public void CanScriptToCode()
        {
            var instance = new User2
            {
                Name = "Trey",
                Address = new Address2
                {
                    AddressLine1 = "123 My Street",
                },
                Personality = new Personality2
                {
                    PersonalityDescriptor = "Intense",
                },
                ByteArray = new byte[] { 1, 2, 3 },
            };

            var code = ITC.Instance.ToCode(instance);

            Approvals.Verify(code);
        }
    }
  
    public class User2
    {
        public string Name { get; set; }
        public Address2 Address { get; set; }
        public Personality2 Personality { get; set; }
        public byte[] ByteArray { get; set; }
    }

    public class Address2
    {
        public string AddressLine1 { get; set; }
    }

    public class Personality2
    {
        public string PersonalityDescriptor { get; set; }
    }
}
