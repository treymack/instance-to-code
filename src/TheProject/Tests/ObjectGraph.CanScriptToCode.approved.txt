new TheProject.Tests.User2()
{
Address = new TheProject.Tests.Address2()
{
AddressLine1 = "123 My Street",
},
Name = "Trey",
Personality = new TheProject.Tests.Personality2()
{
PersonalityDescriptor = "Intense",
},
}